server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java

```diff
@@ -388,16 +330,11 @@ private OnDiskState loadOnDiskState(Path dataPath, DirectoryReader reader) throw
         }
 
         logger.trace("got global metadata, now reading index metadata");
-
-        final Set<String> indexUUIDs = new HashSet<>();
         consumeFromType(searcher, INDEX_TYPE_NAME, bytes ->
         {
             final IndexMetadata indexMetadata = IndexMetadata.fromXContent(XContentFactory.xContent(XContentType.SMILE)
                 .createParser(namedXContentRegistry, LoggingDeprecationHandler.INSTANCE, bytes.bytes, bytes.offset, bytes.length));
             logger.trace("found index metadata for {}", indexMetadata.getIndex());
-            if (indexUUIDs.add(indexMetadata.getIndexUUID()) == false) {
```

Comment:
This change seems unrelated, it's here to catch a case where someone meddled with the metadata and left us with two copies of the metadata for the same index in the same data path. I think we should keep this.